src/Middleware/AcceptAndContentTypeMiddleware.php

```diff
@@ -27,30 +24,28 @@ final class AcceptAndContentTypeMiddleware implements MiddlewareInterface
     private $contentTypeNegotiator;
 
     /**
-     * @var ResponseManagerInterface
+     * @var AcceptAndContentTypeMiddlewareResponseFactoryInterface
      */
-    private $responseManager;
+    private $responseFactory;
 
     public function __construct(
         AcceptNegotiatorInterface $acceptNegotiator,
         ContentTypeNegotiatorInterface $contentTypeNegotiator,
-        ResponseManagerInterface $responseManager
+        AcceptAndContentTypeMiddlewareResponseFactoryInterface $responseFactory
```

Comment:
this is a BC, both arguments should be possible, if the old is given, @trigger_error with user deprecation